### PR TITLE
Chi2 hypothesis test

### PR DESCRIPTION
**Pull request by stucchio**

### CODE REVIEW

**Comment by stucchio** [comment]
Take a look at the test which failed - it has nothing to do with the hypothesis test at all. Might be worth looking into.

**Comment by dlwh** [comment]
it's just yet another place where uncontrolled randomness leads to spurious test failures. I thought I had most of them hammered out, but it's a whack-a-mole kind of thing

File: math/src/main/scala/breeze/stats/hypothesis/package.scala [hunk 1]
```diff
@@ -37,4 +37,64 @@ package object hypothesis {
...
+     */
+    val meanP = (successControl + successVariant).toDouble / (trialsControl + trialsVariant).toDouble
+    val chi2 = (chiSquaredTerm(meanP * trialsControl, successControl) + chiSquaredTerm((1-meanP) * trialsControl, trialsControl - successControl)
+              + chiSquaredTerm(meanP * trialsVariant, successVariant) + chiSquaredTerm((1-meanP) * trialsVariant, trialsVariant - successVariant))
+    val pVal = 1.0 - Gamma(0.5, 2.0).cdf(chi2)
+    Chi2Result(chi2, pVal)
+  }
+
+  def chi2Test(control: (Long, Long), trials: Seq[(Long, Long)]): Seq[Chi2Result] = {
+    /*
```
Comment (dlwh) [inline math/src/main/scala/breeze/stats/hypothesis/package.scala]: any reason why these aren't just scaladoc comments?

**Comment by dlwh** [comment]
if you'd promote your comments to scaladoc, lgtm

File: math/src/main/scala/breeze/stats/hypothesis/package.scala (hunk 1 above)
Comment (stucchio) [inline math/src/main/scala/breeze/stats/hypothesis/package.scala]: Just an oversight. Updated.

**Comment by dlwh** [comment]
fix merge conflicts?

**Comment by stucchio** [comment]
Done

**Comment by dlwh** [comment]
thanks!

**Pull request merged**